controller/src/main/java/com/softserve/borda/controllers/AuthController.java

```diff
@@ -31,6 +31,9 @@
     @PostMapping("/register")
     public ResponseEntity<AuthResponse> registerUser(@RequestBody @Valid RegistrationRequest registrationRequest)
             throws CustomAuthenticationFailedException {
+        if (userService.existsUserByUsername(registrationRequest.getUsername())) {
+            throw new CustomAuthenticationFailedException("Username: [" + registrationRequest.getUsername() + "] is already taken");
```

Comment:
I think, it is better to just return bad request status with detailed message here rather than throw exception.